addons/hr_holidays/hr_holidays.py

```diff
@@ -596,9 +596,16 @@ def _leaves_count(self, cr, uid, ids, field_name, arg, context=None):
         date_begin = date.today().replace(day=1)
         date_end = date_begin.replace(day=calendar.monthrange(date_begin.year, date_begin.month)[1])
         for employee_id in ids:
-            leaves = Holidays.search_count(cr, uid, [('employee_id', '=', employee_id), ('type', '=', 'remove')], context=context)
-            approved_leaves = Holidays.search_count(cr, uid, [('employee_id', '=', employee_id), ('type', '=', 'remove'), ('date_from', '>=', date_begin.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('date_from', '<=', date_end.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('state', '=', 'validate'), ('payslip_status', '=', False)], context=context)
-            res[employee_id] = {'leaves_count': leaves, 'approved_leaves_count': approved_leaves}
+            leaves = Holidays.read_group(cr, uid, [('employee_id', '=', employee_id), ('holiday_status_id.limit', '=', False)], fields=['number_of_days'], groupby=[], context=context)
+            leaves_to_process = Holidays.search_count(cr, uid, [('employee_id', '=', employee_id), ('type', '=', 'remove'), ('date_from', '>=', date_begin.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('date_from', '<=', date_end.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)), ('payslip_status', '=', False)], context=context)
+            res[employee_id] = {'leaves_count': leaves[0]['number_of_days'], 'leaves_to_processe_count': leaves_to_process}
+        return res
+
+    def _show_approved_remaining_leave(self, cr, uid, ids, name, args, context=None):
+        res={}
+        employee= self.browse(cr, uid, ids, context=context)
```

Comment:
This won't work in batch. Remember you receive ids that can be a list containing more than one element. Here you should do
 - if user is hr: return a dict with all ids to true
- check user's employee, set this id to true, other to false